shared/parser.rs

```diff
@@ -127,21 +124,17 @@ pub fn parse_str(mut input: &str) -> Result<[u8; 16], Error> {
                         digit
                     };
 
-                    return Err(ErrorKind::InvalidGroupLength {
-                        expected: ExpectedLength::Exact(GROUP_LENS[group]),
-                        found: found as usize,
+                    return Err(Error::group_length(
+                        ExpectedLength::Exact(GROUP_LENS[group]),
+                        found as usize,
                         group,
-                    }
-                    .into());
+                        start,
+                    ));
                 }
                 _ => {
-                    return Err(ErrorKind::InvalidCharacter {
-                        expected: "0123456789abcdefABCDEF-",
-                        found: input[i_char..].chars().next().unwrap(),
-                        index: i_char,
-                        urn: UrnPrefix::Optional,
-                    }
-                    .into());
+                    // let found = input[i_char..].chars().next().unwrap();
+                    // let found = char::from(chr);
```

Comment:
Did we mean to keep these comments here?

Reply:
Nope that was an accident